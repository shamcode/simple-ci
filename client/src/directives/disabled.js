export default class {
    constructor() {
        this.node = null;
    }

    bind( node ) {
        this.node = node;
    }

    unbind() {
        this.node = null;
    }

    update( value ) {
        this.node.disabled = value;
    }
}
