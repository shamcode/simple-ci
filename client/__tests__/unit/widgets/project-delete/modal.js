import Modal from '../../../../src/widgets/project-delete/modal';
import disabled from '../../../../src/directives/disabled';
import directives from '../../../../src/directives/event-listener';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const meta = renderer( Modal, {
        directives: {
            disabled,
            ...directives
        },
        project: {
            id: '1',
            name: 'Test'
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
