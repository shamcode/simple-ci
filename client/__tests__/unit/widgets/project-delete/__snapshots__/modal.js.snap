// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "ID": "widget",
  "constructorOptions": Object {
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "onsubmit": [Function],
    },
    "project": Object {
      "id": "1",
      "name": "Test",
    },
  },
  "html": "
<div class=\\"modal\\">
  <div class=\\"header\\"> Delete project </div>
  <div class=\\"body\\"> Do you want delete project \\"Test\\"? </div>
  <div class=\\"footer\\">
    <button class=\\"ok danger\\"> Confirm </button>
    <button class=\\"cancel\\"> Cancel </button>
  </div>
</div>
",
  "nodes": Array [
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "dataSaving": false,
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "onsubmit": [Function],
    },
    "onCancel": [Function],
    "onOk": [Function],
    "project": Object {
      "id": "1",
      "name": "Test",
    },
    "renderAsync": false,
    "renderDependence": Array [],
    "types": Array [],
  },
}
`;
